lib/hotel.rb

```diff
@@ -0,0 +1,217 @@
+require_relative 'room'
+require_relative 'reservation'
+require_relative 'dates'
+
+module Hotels
+  # Contains the methods related to reserving rooms and viewing reservations
+  class Hotel
+    attr_reader :rooms, :reservations, :blocks
```

Comment:
Reading through the methods in this class, it seems you've really tried to take the Single Responsibility Principal to heart. I appreciate your dedication to making sure no method is doing too much.

That being said, it is possible to go too far the other direction, to break your logic out into too many small methods. When this happens it ends up being hard to follow the thread of what's going on, because you have to jump up and down all over the file to find the next piece. The code you've written is starting to get to that point. 

Part of this is just experience - the more times you turn a problem spec into working code, the more you'll get a feel for what's the appropriate level of decomposition. However, it's definitely something you can try to be conscious about as we go forward.

For an example of code that gets closer to this balance, [you might be interested in reviewing our reference implementation](https://github.com/droberts-ada/hotel/blob/dpr/solution/lib/hotel.rb).